internal/rukpak/convert/registryv1.go

```diff
@@ -53,92 +53,73 @@ func RegistryV1ToPlain(rv1 fs.FS, installNamespace string, watchNamespaces []str
 		if err != nil {
-			return nil, err
+			return err
 		}
-
-		dec := apimachyaml.NewYAMLOrJSONDecoder(bytes.NewReader(fileData), 1024)
-		for {
-			obj := unstructured.Unstructured{}
-			err := dec.Decode(&obj)
-			if errors.Is(err, io.EOF) {
-				break
-			}
-			if err != nil {
-				return nil, fmt.Errorf("read %q: %v", e.Name(), err)
+		defer func() {
+			if err := manifestFile.Close(); err != nil {
+				l.Error(err, "error closing file", "path", path)
 			}
-			objects = append(objects, &obj)
-		}
-	}
+		}()
 
-	for _, obj := range objects {
-		obj := obj
-		switch obj.GetObjectKind().GroupVersionKind().Kind {
-		case "ClusterServiceVersion":
-			csv := v1alpha1.ClusterServiceVersion{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &csv); err != nil {
-				return nil, err
+		result := resource.NewLocalBuilder().Unstructured().Flatten().Stream(manifestFile, path).Do()
+		if err := result.Err(); err != nil {
+			return err
+		}
+		if err := result.Visit(func(info *resource.Info, err error) error {
+			if err != nil {
+				return err
 			}
-			reg.CSV = csv
-		case "CustomResourceDefinition":
-			crd := apiextensionsv1.CustomResourceDefinition{}
-			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &crd); err != nil {
-				return nil, err
+			switch info.Object.GetObjectKind().GroupVersionKind().Kind {
+			case "ClusterServiceVersion":
+				csv := v1alpha1.ClusterServiceVersion{}
+				if err := runtime.DefaultUnstructuredConverter.FromUnstructured(info.Object.(*unstructured.Unstructured).Object, &csv); err != nil {
+					return err
+				}
+				reg.CSV = csv
+			default:
+				reg.Others = append(reg.Others, *info.Object.(*unstructured.Unstructured))
 			}
-			reg.CRDs = append(reg.CRDs, crd)
-		default:
-			reg.Others = append(reg.Others, *obj)
+			return nil
+		}); err != nil {
+			return fmt.Errorf("error parsing objects in %q: %v", path, err)
 		}
+		return nil
+	}); err != nil {
+		return nil, err
 	}
 
 	plain, err := Convert(reg, installNamespace, watchNamespaces)
 	if err != nil {
 		return nil, err
 	}
 
-	var manifest bytes.Buffer
+	chrt := &chart.Chart{Metadata: &chart.Metadata{}}
 	for _, obj := range plain.Objects {
-		yamlData, err := yaml.Marshal(obj)
+		jsonData, err := json.Marshal(obj)
 		if err != nil {
 			return nil, err
 		}
-		if _, err := fmt.Fprintf(&manifest, "---\n%s\n", string(yamlData)); err != nil {
-			return nil, err
-		}
-	}
-
-	now := time.Now()
-	plainFS := fstest.MapFS{
-		".": &fstest.MapFile{
-			Data:    nil,
-			Mode:    fs.ModeDir | 0755,
-			ModTime: now,
-		},
-		"manifests": &fstest.MapFile{
-			Data:    nil,
-			Mode:    fs.ModeDir | 0755,
-			ModTime: now,
-		},
-		"manifests/manifest.yaml": &fstest.MapFile{
-			Data:    manifest.Bytes(),
-			Mode:    0644,
-			ModTime: now,
-		},
+		hash := sha256.Sum256(jsonData)
+		chrt.Templates = append(chrt.Templates, &chart.File{
+			Name: fmt.Sprintf("object-%x.json", hash[0:8]),
```

Comment:
Is this enough of the hash? Should it be longer?

Reply:
We can make it longer, but I didn't change that aspect of this code. The name is the same as before except for `.json` instead of `.yaml`.